### PR TITLE
feat(code): Hardpoint Empty Label clarification

### PR DESCRIPTION
**Pull request by Zitchas**
This PR is a work in progress intended to replace the `[Empty]` labels with more descriptive ones. 

This PR simply expands the `[Empty]` slot labels to ones that specify exactly what they are, as shown here. This also makes the text of those empty labels dimmer so as to not distract from the occupied hardpoints and to help convey at a glance empty vs full slots.

2024-10-30 Labels too far down v03 fixed 4gun Aerie

### CODE REVIEW

**Comment by Zitchas** [comment]
This is pretty much done and ready to go.

I'm curious as to whether or not we could dim this down a bit, though. Just a touch.

**Review by TheGiraffe3 — APPROVED** [review]

**Comment by Zitchas** [comment]
OK, so one notable change: Whereas the labels normally swap between "dim" (the normal color here) and "bright" when hovering; the empty slots now swap between "dimmer" and "dim" (when hovering). So now the eye is naturally drawn to the full slots, and the empty ones don't demand equal attention. I think this improves the UI and the usability, as well as reinforcing the information being conveyed.

2024-10-29 Mammoth with Pylons  empty labels and dim-dimmer

**Comment by TheGiraffe3** [comment]
That'll make it much easier!

**Comment by Zitchas** [comment]
Currently looking into that gap that happens after the last row of pylons. I would like to replicate it after the last row of guns and the last row of turrets (since it is possible, depending on ship definition, to have turrets listed before guns and/or pylons. That way there should always be a gap between hardpoint types.

edit: I have most of the code figured out to do this, but will have to pause for a bit before I can continue with it. (Thanks to VitalChip for pointers and explaining how `count[0][0]` works, among other things)

File: source/HardpointInfoPanel.cpp [hunk 1]
```diff
@@ -414,7 +416,16 @@ void HardpointInfoPanel::DrawWeapons(const Rectangle & bounds)
 			isRight = stayRight;
 			stayRight = !stayRight;
 		}
-		++count[isRight][hardpoint.IsTurret()];
+		// Check to see if the hardpoint is a gun, pylon, or turret
```
Comment (TheGiraffe3) [inline source/HardpointInfoPanel.cpp]:
```suggestion
		// Check to see if the hardpoint is a gun, pylon, or turret.
```

File: source/HardpointInfoPanel.cpp [hunk 2]
```diff
@@ -434,14 +445,19 @@ void HardpointInfoPanel::DrawWeapons(const Rectangle & bounds)
 	// Figure out how tall each part of the weapon listing will be.
 	int gunRows = max(count[0][0], count[1][0]);
 	int turretRows = max(count[0][1], count[1][1]);
-	// If there are both guns and turrets, add a gap of ten pixels.
-	double height = 20. * (gunRows + turretRows) + 10. * (gunRows && turretRows);
+	int pylonRows = max(count[0][2], count[1][2]);
+	// Add a gap of 10 between different types of mounts.
+	// This calculates the total height required for all the hardpoint labels
```
Comment (TheGiraffe3) [inline source/HardpointInfoPanel.cpp]:
```suggestion
	// This calculates the total height required for all the hardpoint labels.
```

**Comment by Zitchas** [comment]
2024-10-30 Labels too far down v03 fixed Bulk Freighter

Alright, with much help from Vitalchip, the spacing and positioning of the labels is now working properly.

**Pull request merged**